File: 프로그래머스/unrated/181851. 전국 대회 선발 고사/전국 대회 선발 고사.js
function solution(rank, attendance) {
    let answer = 0;
    const participants = rank.filter((el, i) => attendance[i])
                            .sort((a, b) => a - b)
                            .slice(0, 3);
    let score = 10000;
    participants.forEach(participant => {
        answer += rank.indexOf(participant) * score;
        score /= 100;
    })
    
    return answer;
}

File: 프로그래머스/unrated/181851. 전국 대회 선발 고사/전국 대회 선발 고사.java
import java.util.*;

class Solution {
    public int solution(int[] rank, boolean[] attendance) {
        HashMap<Integer, Integer> map = new HashMap<>();
        List<Integer> result = new ArrayList<>();
        
        for (int i = 0; i < rank.length; i++) {
            if (attendance[i]) {
                result.add(rank[i]);
                map.put(rank[i], i);
            }
        }
        
        Collections.sort(result);
        
        return 10000 * map.get(result.get(0)) + 100 * map.get(result.get(1)) + map.get(result.get(2));
    }
}